src/util/check.h

```diff
@@ -42,4 +42,22 @@ class NonFatalCheckError : public std::runtime_error
         }                                                         \
     } while (false)
 
+#if defined(NDEBUG)
+#error "Cannot compile without assertions!"
+#endif
+
+/** Helper for Assert(). TODO remove in C++14 and replace `decltype(get_pure_r_value(val))` with `T` (templated lambda) */
+template <typename T>
+T get_pure_r_value(T&& val)
+{
+    return std::forward<T>(val);
+}
+
+/** Identity function. Abort if the value compares equal to zero */
```

Comment:
Not sure what is meant by *if the value compares equal to zero*.

If you retouch, s/undersirable/undesirable/ a few lines above.

Reply:
> Not sure what is meant by if the value compares equal to zero.

```cpp
if (value == 0) std::abort

Reply:
thx for the typo spot. Fixed.